core-tests/shared/src/test/scala/zio/RefSpec.scala

```diff
@@ -64,10 +64,125 @@ object RefSpec
                        case Changed => ("closed", Closed)
                      }
           } yield assert(value1, equalTo("changed")) && assert(value2, equalTo("closed"))
+        },
+        testM("map") {
+          for {
+            ref    <- Ref.make[State](Active)
+            refInt = ref.map(toInt)
+            value  <- refInt.get
+          } yield assert(value.get, equalTo(1))
+        },
+        testM("contramap") {
+          for {
+            ref <- Ref.make[State](Active)
+            refInt = ref.contramap { i: Int =>
+              fromInt(i).get
+            }
+            _     <- refInt.set(2)
+            value <- refInt.get
+          } yield assert(value, equalTo[State](Changed))
+        },
+        testM("contramap is sound") {
+          for {
+            ref                    <- Ref.make(1)
+            refStringInt           = ref.contramap((_: String).toInt)
+            refString: Ref[String] = refStringInt.map(_.toString)
+            value                  <- refString.update(_.trim())
+          } yield assert(value, equalTo("1"))
         }
       )
     )
 
+// def e9 =
```

Comment:
Looks like you can delete `e9` and `e10` and move the others in the suite above?